assets/scripts/script.js

```diff
@@ -62,5 +62,18 @@
       });
     }
 
+    /**
+     * Modifies data-srcset of product gallery first image on variation change.
+     *
+     * lightGallery gets the srcset from the anchor element that wraps the
+     * image. On variation change, WooCommerce changes the srcset of the first
+     * product image with the variation one, but not the srcset of the wrapping
+     * anchor element.
+     */
+    $('.variations_form').on('woocommerce_variation_has_changed', function() {
+      const firstImage = $('.woocommerce-product-gallery__image img').eq(0);
+      firstImage.parent().attr('data-srcset', firstImage.attr('srcset'));
```

Comment:
Should we modify the parent if it's not the expected?
```suggestion
      firstImage.parent('[data-srcset]').attr('data-srcset', firstImage.attr('srcset'));
```
Or even account for potential other markup structures?
```suggestion
      firstImage.parents('[data-srcset]').attr('data-srcset', firstImage.attr('srcset'));
```
Also, is there always a first image?
```suggestion
      if (firstImage.length) {
        firstImage.parents('[data-srcset]').attr('data-srcset', firstImage.attr('srcset'));
      }
```